real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/LineByLineSensorPlayer.java

```diff
@@ -0,0 +1,112 @@
+package com.roboclub.robobuggy.simulation;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.roboclub.robobuggy.main.RobobuggyConfigFile;
+import com.roboclub.robobuggy.main.RobobuggyLogicNotification;
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Date;
+
+/**
+ *
+ * @author Trevor Decker
+ * a tool for reading log files line by line
+ *
+ */
+public class LineByLineSensorPlayer {
+
+	/**
+	 * Constructor for sensor player
+	 * @param filePath the file to play
+	 * @param playBackSpeed the playback sped
+	 */
+	public LineByLineSensorPlayer(String filePath,double playBackSpeed) {
+		//open up the log file 
+
+		 new RobobuggyLogicNotification("initializing the SensorPlayer", RobobuggyMessageLevel.NOTE);
+		 Thread thread = new Thread(){
+			    public void run(){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method run() has an NPath complexity of 642](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2684047992/issues/source?bid=2938665&fileBranchId=3246677#l36)